packages/astro/src/runtime/client/hmr.ts

```diff
@@ -17,6 +14,27 @@ if (import.meta.hot) {
 				root.innerHTML = current?.innerHTML;
 			}
 		}
-		diff(document, doc);
+		return diff(document, doc);
+	}
+	async function updateAll(files: any[]) {
+		let hasAstroUpdate = false;
+		for (const file of files) {
+			if (file.acceptedPath.endsWith('.astro')) {
+				hasAstroUpdate = true;
+				continue;
+			}
+			if (file.acceptedPath.includes('vue&type=style')) {
+				const link = document.querySelector(`link[href="${file.acceptedPath}"]`);
+				if (link) {
+					link.replaceWith(link.cloneNode(true));
+				}
+			}
+		}
+		if (hasAstroUpdate) {
+			return updatePage()
+		}
```

Comment:
If an `.astro` file has been updated, we trigger our custom HMR here. This is the same event as before, just handled through Vite's built-in logic.